usr/src/uts/armv8/sys/platmod.h

```diff
@@ -33,27 +33,23 @@ extern "C" {
 #include <sys/types.h>
 #include <sys/promif.h>
 
+/*
+ * Called in mp_startup.c from init_cpu_info.
+ */
 extern uint64_t plat_get_cpu_clock(int cpu_no);
```

Comment:
We do call this.  Does it ever really succeed?  I think at the moment we always just return made-up numbers because the actual devicetree checks don't work

Reply:
In qemu-virt it cannot succeed, because even if it found the CPU device node the node never has the `clock-frequency` property.

Reply:
Right.  Neither (I think?) does raspberry pi

Reply:
Oh wow, I think you're right, and right because of what looks like a simple mistake.

Here's the calcite that kicks off the lookup of the CPU clock frequency through the VC: https://github.com/richlowe/illumos-gate/blob/arm64-gate/usr/src/uts/armv8/rpi4/io/rpi4.c#L72

... and here's where it's rejected: https://github.com/richlowe/illumos-gate/blob/arm64-gate/usr/src/uts/armv8/rpi4/io/rpi4.c#L257

I'll look into this once I'm back in the UK. It should be easily fixable.